indico_hub/server.py

```diff
@@ -35,11 +33,84 @@ def _openapi(test, as_json, host, port):
         # TODO: Register all the exposed view functions here.
         #       Don't worry about this at the beginning though!
         # spec.path(view=...)
-
+        spec.path(view=register)
+        spec.path(view=update_instance)
+        spec.path(view=get_instance)
         if as_json:
             print(json.dumps(spec.to_dict()))
         else:
             print(spec.to_yaml())
 
 
 # TODO: Implement the API endpoints here
+@api.route('/api/instance', methods=['POST'])
+@use_kwargs(ValidationSchema, location='json')
+def register(url, contact, email, organization):
+    """
+    mock function for registering an instance to the database
+    ---
+    parameters:
+        Instance.get_json()
+    responses:
+          201:
+            description: instance registered
+          404:
+            description: instance is already registered
+    """
+    inst = Instance().query.filter_by(url=url).first()
+    if inst:
+        current_app.logger.exception('register: This instance is already registered')
+        abort(401, description='BAD_REQUEST')
+    inst = Instance(
+        url=url,
+        contact=contact,
+        email=email,
+        organization=organization,
+    )
+    db.session.add(inst)
+    db.session.commit()
+    return InstanceSchema().dumps(inst), 201
+
+
+@api.route('/api/instance/<string:url>', methods=['PATCH', 'POST'])
+@use_kwargs(UpdateInstance, location='json')
+def update_instance(url, **kwargs):
+    """
+    updates information regarding the instance
+    ---
+    parameters:
+        -enabled
+        -url
+        -contact
+        -email
+        -organization
+    responses:
+        200: updated instance
+        400: missing argument | bad_url | Instance doesn't exist
+    """
+    if url is None:
+        abort(400, description='BAD_URL')
+    inst = Instance().query.filter_by(url=url).first()
```

Comment:
query is a class attribute, you do not need to instantiate the model to use it

Reply:
solved